packages/stream_chat_flutter/lib/src/message_modal/message_action_confirmation_modal.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+import 'package:stream_chat_flutter/src/misc/adaptive_dialog_action.dart';
+import 'package:stream_chat_flutter/src/theme/stream_chat_theme.dart';
+
+/// {@template streamMessageActionConfirmationModal}
+/// A confirmation modal dialog for message actions in Stream Chat.
+///
+/// This widget creates a platform-adaptive confirmation dialog that can be used
+/// when a user attempts to perform an action on a message that requires
+/// confirmation (like delete, flag, etc).
+///
+/// The dialog presents two options: cancel and confirm, with customizable text
+/// for both actions. The confirm action can be styled as destructive for
+/// actions like deletion.
+///
+/// Example usage:
+///
+/// ```dart
+/// showDialog(
+///   context: context,
+///   builder: (context) => StreamMessageActionConfirmationModal(
+///     title: Text('Delete Message'),
+///     content: Text('Are you sure you want to delete this message?'),
+///     confirmActionTitle: Text('Delete'),
+///     isDestructiveAction: true,
+///   ),
+/// ).then((confirmed) {
+///   if (confirmed == true) {
+///     // Perform the action
+///   }
+/// });
+/// ```
+/// {@endtemplate}
+class StreamMessageActionConfirmationModal extends StatelessWidget {
+  /// Creates a message action confirmation modal.
+  ///
+  /// The [cancelActionTitle] defaults to a Text widget with 'Cancel'.
+  /// The [confirmActionTitle] defaults to a Text widget with 'Confirm'.
+  /// Set [isDestructiveAction] to true for actions like deletion that should
+  /// be highlighted as destructive.
+  const StreamMessageActionConfirmationModal({
+    super.key,
+    this.title,
+    this.content,
+    this.cancelActionTitle = const Text('Cancel'),
+    this.confirmActionTitle = const Text('Confirm'),
```

Comment:
I would make these nullable, so the default implementations can use localizations.

Reply:
Should be fine, we also have to support a additional `Confirm` text translations for all languages